Build script broken by cargo workspace

As described in the comments of #213, looks like the `build-rust-xcode.sh` script is looking for the binary in the `xi-editor/rust/syntect-plugin/target`, but since there's [now a Cargo workspace](https://github.com/google/xi-editor/pull/730) in the `xi-editor/rust/Cargo.toml`, the `target` folder is now at `xi-editor/rust/target`.

I've just tried building this for the first time and run into the same error:

```
   Compiling xi-syntect-plugin v0.0.0 (file:///Users/chris/xi-mac.git/xi-editor/rust/syntect-plugin)
    Finished release [optimized] target(s) in 117.71 secs
fatal error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: can't open input file: target/x86_64-apple-darwin/release/xi-syntect-plugin (No such file or directory)

** BUILD FAILED **


The following build commands failed:
	ExternalBuildToolExecution xicore
```

--- a/build-rust-xcode.sh
+++ b/build-rust-xcode.sh
@@ -23,7 +23,8 @@ set -e
 
 function build_target () {
     TARGET_NAME="$1"
-    cd "${SRCROOT}/xi-editor/$2"
+    CARGO_PROJECT_FLAG="$2"
+    cd "${SRCROOT}/xi-editor/rust"
     if [[ ${ACTION:-build} = "build" ]]; then
         if [[ $PLATFORM_NAME = "" ]]; then
             # default for building with xcodebuild
@@ -58,7 +59,7 @@ function build_target () {
             if [[ $RUST_ARCH = "arm64" ]]; then
                 RUST_ARCH="aarch64"
             fi
-            cargo build $RUST_CONFIGURATION_FLAG --target "${RUST_ARCH}-apple-${RUST_TARGET_OS}"
+            cargo build $RUST_CONFIGURATION_FLAG $CARGO_PROJECT_FLAG --target "${RUST_ARCH}-apple-${RUST_TARGET_OS}"
             EXECUTABLES+=("target/${RUST_ARCH}-apple-${RUST_TARGET_OS}/${RUST_CONFIGURATION}/${TARGET_NAME}")
         done
 
@@ -70,8 +71,8 @@ function build_target () {
     fi
 }
 
-build_target xi-core rust
-build_target xi-syntect-plugin rust/syntect-plugin
+build_target xi-core ""
+build_target xi-syntect-plugin "-p xi-syntect-plugin"
 
 # move syntect plugin into plugins dir
 mkdir -p "${BUILT_PRODUCTS_DIR}/plugins/syntect/bin"

